core/src/main/java/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractor.java

```diff
@@ -56,12 +58,17 @@ public static ResultTrain run(
         // Compute events
         var events = computeEvents(infra, trainPath, trainLength, envelopeWithStops);
 
+        // Compute energy consumed
+        var envelopePath = EnvelopeTrainPath.from(trainPath);
```

Comment:
We need an EnvelopePath for the method getMechanicalEnergyConsumed, and trainPath is a TrainPath, so we need to convert it. We can't find the envelopetrainpath you're mentionning above :/

Reply:
When it's called from `StandaloneSim` we have an `EnvelopeTrainPath` available, but we also call it from STDCM, and it's not readily available there

Reply:
In STDCM context, no `envelopePath` is computed before calling the `run` method from `ScheduleMetaDataExtractor`, so it would be quite a pain to modify everything.